src/Twilio/Security/RequestValidator.cs

```diff
@@ -81,27 +129,16 @@ public bool ValidateBody(string rawBody, string expected)
             {
                 dict.Add(k, col[k]);
             }
+
             return dict;
         }
 
-        private string GetValidationSignature(string url, IDictionary<string, string> parameters)
+        private string GetValidationSignature(string urlWithParameters)
         {
-            var b = new StringBuilder(url);
-            if (parameters != null)
-            {
-                var sortedKeys = new List<string>(parameters.Keys);
-                sortedKeys.Sort(StringComparer.Ordinal);
-
-                foreach (var key in sortedKeys)
-                {
-                    b.Append(key).Append(parameters[key] ?? "");
-                }
-            }
-
-            var hash = _hmac.ComputeHash(Encoding.UTF8.GetBytes(b.ToString()));
+            byte[] hash = _hmac.ComputeHash(Encoding.UTF8.GetBytes(urlWithParameters));
```

Comment:
To re-discussed the thread safety discussion based on https://github.com/twilio/twilio-csharp/pull/598..

_Is_ `RequestValidator` meant to be thread safe? If it is, then the fields make this problematic. That linked PR largely fixes the problem, but could be optimized some.

If we are looking for performance opportunities here, I think quite a bit could be gained if we had a `net6.0` target in here and did some conditional compilation.

If we had a net6.0 target, we could do `#if NET6_0_OR_GREATER` and use the hash and HMAC one shots, and change the cryptographic primitive fields to locals. That would get you better performance and thready safety for .NET 6+. It would mean less performance for "not" .NET 6, but you'd get thread safety using local crypto primitives.

There are many other optimization opportunities we can eke out here if we have a net6.0 target with `Span` and some careful use of stackalloc or rented buffers. If you're willing to take a .NET 6 TFM in the project, you can merge this PR as-is and I am happy to take a pass at squeezing out all of the performance, if you would like.

Reply:
Results from using the one-shots:

Method | Mean | Error | StdDev | Gen0 | Gen1 | Allocated
-- | -- | -- | -- | -- | -- | --
OriginalUnhappyPath | 14.902 us | 0.0598 us | 0.0467 us | 1.9836 | 0.2441 | 12.27 KB
CurrentUnhappyPath | 10.830 us | 0.0574 us | 0.0509 us | 1.3885 | - | 8.52 KB
OriginalHappyPath | 8.756 us | 0.1750 us | 0.1551 us | 1.6327 | 0.2289 | 10.02 KB
CurrentHappyPath | 2.625 us | 0.0239 us | 0.0199 us | 0.6256 | - | 3.85 KB

I have it in a [separate branch](https://github.com/Swimburger/twilio-csharp/tree/optimize-request-validation-thread-safe) for now. I'll ask around internally about the .NET 6.0 TFM next week.

Reply:
@benmccallum PR has been merged in and I added the .NET 6 one-shots. There may be a less messy way of writing the code.
If there's a way to improve it with spans, let's try!